src/xwayland.rs

```diff
@@ -145,24 +190,36 @@ impl XWayland {
         })
     }
 
-    /// Listen for window created events on the root window
-    pub fn listen_for_window_created(
+    /// Listen for window lifecycle events on the root window
+    pub fn listen_for_window_lifecycle(
```

Comment:
Changing this function name will cause an API breaking change. We should instead create a new function, and mark the old one with the deprecated macro:

```rust
#[deprecated(since="0.1.0", note="please use `listen_for_window_lifecycle` instead")]
pub fn listen_for_window_created()
```